src/App.vue

```diff
@@ -1,32 +1,53 @@
 <template>
-  <div id="app">
-    <div id="nav">
-      <router-link to="/">Home</router-link> |
-      <router-link to="/about">About</router-link>
-    </div>
-    <router-view/>
+  <div>
+      <router-view />
   </div>
 </template>
 
-<style lang="scss">
-#app {
-  font-family: Avenir, Helvetica, Arial, sans-serif;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-  text-align: center;
-  color: #2c3e50;
-}
+<script>
+import axios from 'axios'
+import store from './store'
 
-#nav {
-  padding: 30px;
-
-  a {
-    font-weight: bold;
-    color: #2c3e50;
-
-    &.router-link-exact-active {
-      color: #42b983;
+export default {
+  name: 'App',
+  data () {
+    return {
+      restaurants: [],
+      loading: false
+    }
+  },
+  methods: {
+    // eslint-disable-next-line
+    fetchRestaurants: function() {
+      this.loading = true
+      const config = {
+        'user-key': process.env.VUE_APP_API_KEY,
+        Accept: 'application/json'
+      }
+      axios.get('https://developers.zomato.com/api/v2.1/search?entity_id=262&entity_type=city&count=20&collection_id=1&sort=rating&order=desc', { headers: config })
+        .then((response) => {
+          this.restaurants = response.data.restaurants
+          this.loading = false
+          store.restaurants = this.restaurants
+        })
+        .catch((error) => {
+          console.log(error)
+        })
     }
+  },
+  created () {
+    this.fetchRestaurants()
```

Comment:
It's interesting how you made it a part of `App.vue` and not `Home.vue`, splitting the responsibilities between "action" App and "presentation" Home. Thanks to that, even if I manually paste some particular `/restaurant/:id`, I still get to download the list and therefore the whole thing works --- neat!